reactive/flume_hdfs.py

```diff
@@ -0,0 +1,42 @@
+from charms.reactive import when, when_not
+from charms.reactive import set_state, remove_state
+from charmhelpers.core import hookenv
+from charms.flume import Flume
+from charms.hadoop import get_dist_config
```

Comment:
With get_dist_config I was getting the following exception:
unit-fhf-0[11761]: 2016-02-25 13:28:29 INFO unit.fhf/0.juju-log server.go:268 hadoop:10: Invoking reactive handler: reactive/flume_hdfs.py:25:configure_flume
unit-fhf-0[11761]: 2016-02-25 13:28:29 INFO unit.fhf/0.hadoop-relation-changed logger.go:40 Traceback (most recent call last):
unit-fhf-0[11761]: 2016-02-25 13:28:29 INFO unit.fhf/0.hadoop-relation-changed logger.go:40   File "/var/lib/juju/agents/unit-fhf-0/charm/hooks/hadoop-relation-changed", line 19, in <module>
unit-fhf-0[11761]: 2016-02-25 13:28:29 INFO unit.fhf/0.hadoop-relation-changed logger.go:40     main()
unit-fhf-0[11761]: 2016-02-25 13:28:29 INFO unit.fhf/0.hadoop-relation-changed logger.go:40   File "/usr/local/lib/python3.4/dist-packages/charms/reactive/__init__.py", line 67, in main
unit-fhf-0[11761]: 2016-02-25 13:28:29 INFO unit.fhf/0.hadoop-relation-changed logger.go:40     bus.dispatch()
unit-fhf-0[11761]: 2016-02-25 13:28:29 INFO unit.fhf/0.hadoop-relation-changed logger.go:40   File "/usr/local/lib/python3.4/dist-packages/charms/reactive/bus.py", line 405, in dispatch
unit-fhf-0[11761]: 2016-02-25 13:28:29 INFO unit.fhf/0.hadoop-relation-changed logger.go:40     _invoke(other_handlers)
unit-fhf-0[11761]: 2016-02-25 13:28:29 INFO unit.fhf/0.hadoop-relation-changed logger.go:40   File "/usr/local/lib/python3.4/dist-packages/charms/reactive/bus.py", line 388, in _invoke
unit-fhf-0[11761]: 2016-02-25 13:28:29 INFO unit.fhf/0.hadoop-relation-changed logger.go:40     handler.invoke()
unit-fhf-0[11761]: 2016-02-25 13:28:29 INFO unit.fhf/0.hadoop-relation-changed logger.go:40   File "/usr/local/lib/python3.4/dist-packages/charms/reactive/bus.py", line 269, in invoke
unit-fhf-0[11761]: 2016-02-25 13:28:29 INFO unit.fhf/0.hadoop-relation-changed logger.go:40     self._action(*args)
unit-fhf-0[11761]: 2016-02-25 13:28:29 INFO unit.fhf/0.hadoop-relation-changed logger.go:40   File "/var/lib/juju/agents/unit-fhf-0/charm/reactive/flume_hdfs.py", line 29, in configure_flume
unit-fhf-0[11761]: 2016-02-25 13:28:29 INFO unit.fhf/0.hadoop-relation-changed logger.go:40     flume.configure_flume()
unit-fhf-0[11761]: 2016-02-25 13:28:29 INFO unit.fhf/0.hadoop-relation-changed logger.go:40   File "lib/charms/layer/flume_base.py", line 77, in configure_flume
unit-fhf-0[11761]: 2016-02-25 13:28:29 INFO unit.fhf/0.hadoop-relation-changed logger.go:40     target=self.config_file,
unit-fhf-0[11761]: 2016-02-25 13:28:29 INFO unit.fhf/0.hadoop-relation-changed logger.go:40   File "lib/charms/layer/flume_base.py", line 29, in config_file
unit-fhf-0[11761]: 2016-02-25 13:28:29 INFO unit.fhf/0.hadoop-relation-changed logger.go:40     return self.dist_config.path('flume_conf') / 'flume.conf'
unit-fhf-0[11761]: 2016-02-25 13:28:29 INFO unit.fhf/0.hadoop-relation-changed logger.go:40   File "/usr/local/lib/python3.4/dist-packages/jujubigdata/utils.py", line 102, in path
unit-fhf-0[11761]: 2016-02-25 13:28:29 INFO unit.fhf/0.hadoop-relation-changed logger.go:40     dirs = {name: self.dirs[name]['path'] for name in self.dirs.keys()}
unit-fhf-0[11761]: 2016-02-25 13:28:29 INFO unit.fhf/0.hadoop-relation-changed logger.go:40 AttributeError: 'DistConfig' object has no attribute 'dirs'
unit-fhf-0[11761]: 2016-02-25 13:28:29 ERROR juju.worker.uniter.operation runhook.go:107 hook "hadoop-relation-changed" failed: exit status 1